exodus_lambda/functions/origin_request.py

```diff
@@ -320,10 +320,28 @@ def response_from_db(self, request, table, uri):
 
             raise err
 
+    def validate_request(self, request):
+        # Validate URI and query string lengths, as those are only elements provided by users.
+        #
+        # For example, see
+        # https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-event-structure.html#example-origin-request
+
+        for key in ["uri", "querystring"]:
+            if key in request and not 1 < len(request[key]) < 2000:
+                exc_info = ValueError("Request validation failed")
+                self.logger.exception(
+                    "%s exceeds length limits: %s",
+                    key,
+                    request[key],
+                    exc_info=exc_info,
+                )
+                raise exc_info
+
     def handler(self, event, context):
         # pylint: disable=unused-argument
 
         request = event["Records"][0]["cf"]["request"]
+        self.validate_request(request)
```

Comment:
I think just raising an exception here is going to give us a 500 error.  It'll be flagged as a lambda execution error.  This is not good for us because in theory we'd want to have all of those alerting us, so we can investigate our bugs.

Can we make it so that a validation failure instead results in a 400 (bad request) response code?